src/core/public/saved_objects/saved_objects_service.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreService } from 'src/core/types';
+import { SavedObjectsClient, SavedObjectsClientContract } from './saved_objects_client';
+
+/**
+ * @public
+ */
+export interface SavedObjectsStart {
+  /** {@link SavedObjectsClient} */
+  client: SavedObjectsClientContract;
+}
+
+export const SavedObjectsService: CoreService<void, SavedObjectsStart> = {
+  setup: async function setup() {},
+  start: async function start({ http }) {
+    return { client: new SavedObjectsClient(http) };
```

Comment:
I'd expect that plugins need this API during `setup` step, when they configure an application

Reply:
I disagree, applications should be accessing this via the ApplicationMountContext.

Reply:
I'm okay with it. just want to understand why

Reply:
@joshdover Do you mean that in `setup` applications will register a mount handler which has access to the `SavedObjectsClient` through the `ApplicationMountContext` so they won't need to directly access the client from CoreSetup?

Reply:
> @joshdover Do you mean that in `setup` applications will register a mount handler which has access to the `SavedObjectsClient` through the `ApplicationMountContext` so they won't need to directly access the client from CoreSetup?

Yes.

> I'm okay with it. just want to understand why. handler context doesn't deprecate the direct access to core API but facilitates access to it

I think we should be trying to limit the number of things plugins do during the setup phase so that Kibana can start up and be handling traffic _fast_. When we expose anything that does async work (Elasticsearch, HTTP clients, etc.) we give plugins more opportunity to do things that slow the system down.

There may be be cases where this is unavoidable, but until we have a strong use case for using SavedObjects during the setup phase, I don't think we should allow it.